core/request.go

```diff
@@ -209,28 +214,57 @@ func SubmitRequest(t []TransferRequest, dstUrl string) error {
 }
 
 // RedirectRequest function to send the request to source
-func RedirectRequest(t *TransferRequest, dstUrl string) error {
-	selectedAgents, index, err := AgentRouter.FindSource(t)
+func RedirectRequest(t *TransferRequest) error {
+	var (
+		selectedAgents []SourceStats
+		index          int
+		err            error
+	)
+	if RouterModel == true {
+		selectedAgents, index, err = AgentRouter.FindSource(t)
+	} else if TransferType == "pull" {
```

Comment:
This else if statement seems to break again test/run.sh and it prohibit request resolution. I think it is mistake and should not be here. Plus, the logic of if statement should be if router do something otherwise do something else. While right now it mix different statements, router and trasfnfer type.